### ISSUE TITLE
It should be possible that .toRaiseError can also accept a regular expression.

### ISSUE THREAD

**Issue by SerkanSipahi** [tstyche/tstyche#359]
### Feature enhancement for `.toRaiseError`

For simple type error messages it is sufficient that `.toRaiseError` takes a string but for more complex error messages this message can become very long which makes it difficult to read or understand in the type tests. Moreover, most of the time, in complex type errors, we are interested in the end of the type error message.

Complex Error Message:

```bash
Argument of type '{ id: "foo"; description: string; _timeoutId: number; loading: false; status: "error"; }' is not assignable to parameter of type 'WithLoader<"foo" | `loader$${string}`>'.
     Type '{ id: "foo"; description: string; _timeoutId: number; loading: false; status: "error"; }' is not assignable to type 'WithLoaderError<"foo" | `loader$${string}`, unknown>'.
       Property 'error' is missing in type '{ id: "foo"; description: string; _timeoutId: number; loading: false; status: "error"; }' but required in type '{ status: "error"; error: unknown[]; }'
```

Actually on what we are interested to know:

```bash
Property 'error' is missing in type '{ id: "foo"; description: string; _timeoutId: number; loading: false; status: "error"; }'
```

Therefore it would be of great advantage if we could pass alongside to a string also a regular expression to `.toRaiseError` 

```ts
expect<'foo bar baz'>().type.toRaiseError(/foo/)
```

**Comment by mrazauskas** [comment]
I have just shipped `tstyche@3.0.0-rc.1` with this feature. Give it a try.

Just to draw your attention: if a string is passed to `.toRaiseError()`, it is interpreted as a substring. In other words, this can be just a part of the error message. I have to explain this better in documentation.

**Comment by SerkanSipahi** [comment]
Oh really. I have not tested this at all. However, I think that the regex feature will still be very useful. Nevertheless, many thanks for your efforts in implementing this.
>  if a string is passed to .toRaiseError(), it is interpreted as a substring

**Comment by SerkanSipahi** [comment]
@mrazauskas It seems that we can pass a list of substrings to .toRaiseError() But how does that work? My first quick tests were not successful.

**Comment by mrazauskas** [comment]
If you ass multiple arguments, those are matched with multiple errors:

https://github.com/tstyche/tstyche/blob/7d51473681bfa1a21af91d0f2ef7a9e219b20803/tests/__fixtures__/api-toRaiseError/__typetests__/toRaiseError-regex.tst.ts#L29-L36

Now I see this is also not documented, but all is simple: the count of errors and args must match, the sequence must match, and the list of args can be any mixture of substrings, error codes or also regular expressions.

**Comment by SerkanSipahi** [comment]
Now I understand. My intension before I saw your example (see above) was how it should work: I can pass a series of substrings which are then matched against the type error message.

**Comment by SerkanSipahi** [comment]
Migrated to v3.0.0-rc.2. It behaves as expected. Thank you.

**Comment by mrazauskas** [comment]
Shipped with [tstyche@3.0.0](https://github.com/tstyche/tstyche/releases/tag/v3.0.0).
